### PR TITLE
Do not update id on fetch (fixes #1)

### LINKED ISSUE #1: Checkout will fail on first run
In the README the user is told to run `fetch` before `checkout`. However the first fetch will fail with the following message:
```Using "mydomain.org" domain configuration…
5 redirections found, fetching details…
Redirections written to ".mydomain.org_ovh-redirections.yaml"
[Errno 2] No such file or directory: 'mydomain.org_ovh-redirections.yaml'
Maybe you missed to run './ovh-redirections checkout' before ?
```
Running the `checkout` command and then the `fetch` command solves the issue. It seems that the software is writing a file and then reopening it, maybe the file was not closed before the read.

### PR DESCRIPTION
**Pull request by gagath**
The `_update_local_ids()` function will try to update the `mydomain.org_ovh-redirections.yaml` file (with no dot) but the fetch function will create the `.mydomain.org_ovh-redirections.yaml` (with a dot) so thats why there is an error.

Wait for checkout in order to update local ids should be enough.

### CODE REVIEW

**Comment by neomilium** [comment]
Updating IDs is required to have a consistent output after 'commit' : as IDs are generated on server side, you need to re-fetch data to your local cache (.mydomain.tld.yaml) and update IDs in your working file (mydomain.tld.yaml).
So, you need to make some improvements to support this case.

Thanks !

**Comment by gagath** [comment]
I did not saw that the `fetch` command was called after each `commit`.

I have changed the behavior of the fetch method: the `_update_local_ids` function will now be called on every fetch expect if the local database was not found. The first call to `fetch` now does not fail.

This questions the error handling at line 116, since not having the local database *is a normal state* according to the README, between the `fetch` and the `checkout` calls.

File: ovh-redirections [hunk 1]
```diff
@@ -71,15 +71,27 @@ def fetch():
...
     with open(remote_db, 'w') as f_r:
         yaml.dump(redirections_dict, f_r, indent=2, default_flow_style=False)
+
     print('Redirections written to "{}"'.format(remote_db))
 
+    # Try to update local IDs if the local database exists
+    if os.path.isfile(local_db):
+        _update_local_ids()
+    else:
+        print("Local database not found, skipping update local IDs...")
```
Comment (neomilium) [inline ovh-redirections]: You should remove this displayed text about "not updating ID" : its more a debug message than user message.

**Comment by neomilium** [comment]
Great, thanks !

Could you squash your commits that are related to fixing ID on first fetch, and keep only 2 commits (one for fixing, one for README improvement) ?

**Comment by gagath** [comment]
I have reworked the pull request in two commits. In the diff we can see two differences:
- Do not update IDs if local database is missing
- Update local IDs on checkout

The first point is correct, but I am not sure about the second one.

File: ovh-redirections [hunk 2]
```diff
@@ -96,6 +106,7 @@ def _update_local_ids():
 
 def checkout():
     shutil.copyfile(remote_db, local_db)
+    _update_local_ids()
```
Comment (neomilium) [inline ovh-redirections]: As the file is copied, why updating IDs ?

**Comment by gagath** [comment]
Updating the local IDs during the `checkout` was not useful indeed. I have removed it and rebased the pull request in two commits as before: one for the code fix and one for the README.

**Comment by neomilium** [comment]
Thanks !

**Pull request merged**